sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -299,7 +314,7 @@ case class ListQuery(
     childOutputs.head.dataType
   }
   override lazy val resolved: Boolean = childrenResolved && plan.resolved && childOutputs.nonEmpty
-  override def nullable: Boolean = false
+  override def nullable: Boolean = plan.schema.fields.exists(_.nullable)
```

Comment:
`ListQuery` should be nullable, because `InSubquery`'s nullability depends on both left values and the subquery. If the subquery has nullable columns, `InSubquery` could be null.

Reply:
I might miss it, but as we don't execute `ListQuery` directly, do we need this?

Reply:
It works in the optimizer. Example:
```
select *
from t1
where (3 in (select a from t2)) is null
``` 

If we don't change this, the filter will be eliminated because `3 in (select a from t2)` is not nullable.

Reply:
Is an existing issue or related to this change?

Reply:
Related to this change, because we rewrite it as join before so the nullability of ListQuery was not used.